math/libfaiss/Portfile

```diff
@@ -24,39 +22,134 @@ long_description    Library for efficient similarity search and clustering of \
-build.env-append    "CFLAGS=-fPIC -m64 -Wall -g -O3 -msse4 -mpopcnt -fopenmp \
-                            -Wno-sign-compare -std=c++11 -I/usr/include/malloc/"
-build.env-append    "LDFLAGS=-g -fPIC -fopenmp"
-build.env-append    SHAREDEXT=dylib
-build.env-append    "SHAREDFLAGS=-Wl,-F. -bundle -undefined dynamic_lookup"
-build.env-append    FAISSSHAREDFLAGS=-dynamiclib
-build.env-append    BLASCFLAGS=-DFINTEGER=int
-build.env-append    "BLASLDFLAGS=-framework Accelerate"
 build.target        libfaiss.a libfaiss.dylib
 
-destroot.target
-destroot.destdir
-destroot.cmd        mkdir -p ${destroot}${prefix}/lib && \
-                    cp libfaiss.a libfaiss.dylib ${destroot}${prefix}/lib && \
-                    mkdir -p ${destroot}${prefix}/include/faiss && \
-                    cp -R ./*.h ${destroot}${prefix}/include/faiss/
-
-variant openblas description {Use OpenBLAS instead of Apple's Accelerate framework} {
-    depends_lib-append  port:OpenBLAS
-    build.env-delete    "BLASLDFLAGS=-framework Accelerate"
-    build.env-append    BLASLDFLAGS=${prefix}/lib/libopenblas.dylib
+if {${name} eq ${subport}} {
+    variant openblas description {Use OpenBLAS instead of Apple's Accelerate framework} {
+        depends_lib-append  port:OpenBLAS
+        build.env-delete    "BLASLDFLAGS=-framework Accelerate"
+        build.env-append    BLASLDFLAGS=${prefix}/lib/libopenblas.dylib
+    }
+
+    default_variants +openblas
+}
+
+# create Python subports
+set python_versions {37 38}
+foreach v ${python_versions} {
+    subport py${v}-${github.project} {
+        PortGroup           python 1.0
+        python.default_version  ${v}
+    }
 }
 
-default_variants +openblas
+# Python bindings for supported Python versions
+if {[string match "py*" ${subport}]} {
+    github.setup        ${github.author} ${github.project} ${github.version} ${github.tag_prefix}
+
+    categories-append   math
+    description         Python ${python.branch} bindings for ${description}
+    long_description    This package provides Python ${python.branch} bindings, \
+                        modules, and scripts.
+
+    depends_build-append \
+                        port:py${python.version}-setuptools \
+                        port:swig-python
+    depends_lib-append  port:${name} \
+                        port:py${python.version}-numpy
+
+    post-extract {
+        # fix destroot commands like `$(PYTHON) setup.py install`
+        reinplace -E \
+            "s|^(\[\[:space:]]+\\\$\\(PYTHON\\)\[\[:space:]]+setup\.py\[\[:space:]]+)(install)|\\1 --no-user-cfg \\2 ${destroot.destdir}|" \
+            ${worksrcpath}/python/Makefile
+    }
+
+    use_configure       yes
+
+    configure.env-append \
+                        PYTHON=${python.bin}
+
+    configure.args-append \
+                        --with-python=${python.bin} \
+                        --with-swig=${prefix}/bin/swig
+
+    pre-build {
+        system -W ${worksrcpath} ${build.cmd}
+    }
```

Comment:
this now builds the library again - we don't want to do that as you already added a dependency on `libfaiss` so it should use the library that was already build by that port instead of building it again

Reply:
The make system requires this: https://github.com/facebookresearch/faiss/blob/master/INSTALL.md

Reply:
I just added a symlink to `libfaiss.a` in `worksdrcpath` and resolved all these issues.